src/Corax/IndexEntry/IndexEntryWriter.cs

```diff
@@ -91,236 +117,254 @@ public void WriteRaw(int field, ReadOnlySpan<byte> binaryValue)
-        ref int fieldLocation = ref _knownFieldsLocations[field];
+        ref int fieldLocation = ref KnownFieldsLocations[field];
         fieldLocation = dataLocation | Constants.IndexWriter.IntKnownFieldMask;
 
         // Write the tuple information. 
-        fixed (byte* bufferPtr = _buffer)
-        {
-            ref var indexEntryField = ref Unsafe.AsRef<IndexEntryFieldType>(bufferPtr + dataLocation);
-
-            indexEntryField = IndexEntryFieldType.Tuple;
-            dataLocation += sizeof(IndexEntryFieldType);
+        ref var indexEntryField = ref Unsafe.AsRef<IndexEntryFieldType>(Unsafe.AsPointer(ref buffer[dataLocation]));
+        indexEntryField = IndexEntryFieldType.Tuple;
+        dataLocation += sizeof(IndexEntryFieldType);
 
-            dataLocation += VariableSizeEncoding.Write(_buffer, longValue, dataLocation);
-            Unsafe.WriteUnaligned(ref _buffer[dataLocation], doubleValue);
-            dataLocation += sizeof(double);
-            dataLocation += VariableSizeEncoding.Write(_buffer, value.Length, dataLocation);
+        dataLocation += VariableSizeEncoding.Write(buffer, longValue, dataLocation);
+        Unsafe.WriteUnaligned(ref buffer[dataLocation], doubleValue);
+        dataLocation += sizeof(double);
+        dataLocation += VariableSizeEncoding.Write(buffer, value.Length, dataLocation);
 
-            // Copy the actual string data. 
-            if (value.Length != 0)
-            {
-                value.CopyTo(_buffer.Slice(dataLocation, value.Length));
-                dataLocation += value.Length;
-            }
-            else
-            {
-                // Signal that we will have to deal with the empties.
-                indexEntryField |= IndexEntryFieldType.Empty;
-            }
-
-            _dataIndex = dataLocation;
+        // Copy the actual string data. 
+        if (value.Length != 0)
+        {
+            value.CopyTo(buffer.Slice(dataLocation, value.Length));
+            dataLocation += value.Length;
+        }
+        else
+        {
+            // Signal that we will have to deal with the empties.
+            indexEntryField |= IndexEntryFieldType.Empty;
         }
+
+        _dataIndex = dataLocation;
     }
 
-    public void Write<TEnumerator>(int field, TEnumerator values, IndexEntryFieldType type = IndexEntryFieldType.Null)
+    public void Write<TEnumerator>(int field, TEnumerator values, IndexEntryFieldType type = IndexEntryFieldType.Null) 
         where TEnumerator : IReadOnlySpanIndexer
     {
         Debug.Assert(field < _knownFields.Count);
-        Debug.Assert(_knownFieldsLocations[field] == Invalid);
+        Debug.Assert(KnownFieldsLocations[field] == Invalid);
+
+        // Calculate the required space. 
+        int requiredSpace = 0;
+        for (int i = 0; i < values.Length; i++)
+            requiredSpace += values[i].Length;
+        requiredSpace += sizeof(IndexEntryFieldType) + 2 * values.Length * sizeof(long) + 4 * sizeof(long);
```

Comment:
I wonder if things like `2 * values.Length * sizeof(long)` and `4 * sizeof(long);` shouldn't be put in a dedicated variable with meaningful name. I'm not very familiar with the details and for me it's very hard to figure out what is it exactly about - so it's harder to spot a possible miscalculation. I think that could be helpful for a reader of this code in the future.

Reply:
Or at least the comment explaining it